src/App.jsx

```diff
@@ -14,33 +15,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function prepairGoods(goods, sortField) {
+  const prepGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortField) {
+    prepGoods.sort((goods1, goods2) => {
+      switch (sortField) {
+        case 'Sort alphabetically':
+          return goods1.localeCompare(goods2);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+        case 'Sort by length':
+          return goods1.length - goods2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return prepGoods;
+}
+
+export const App = () => {
+  const [sortGoods, setSortGoods] = useState('');
+  let visibleGoods = prepairGoods(goodsFromServer, sortGoods);
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.toReversed();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${sortGoods === 'Sort alphabetically' ? '' : 'is-light'}`}
+          onClick={() => {
+            setSortGoods('Sort alphabetically');
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={`button is-success ${sortGoods === 'Sort by length' ? '' : 'is-light'}`}
+          onClick={() => {
+            setSortGoods('Sort by length');
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${reversed ? '' : 'is-light'}`}
+          onClick={() => setReversed(!reversed)}
```

Comment:
to avoid unexpected beheviour let's use functional set state in this case
```suggestion
          onClick={() => setReversed((prevState) => !prevState)}
```